File: app-front/src/services/recovery.service.ts
import api from "./api";

export const changePassword = async (data: { 
  usr_id: number; 
  old_password: string; 
  new_password: string; 
}) => {
  try {
    const res = await api.post("/security/change-password", data);
    return res.data;
  } catch (error: any) {
    return error.response?.data || { result: false, message: "Error al cambiar contraseña" };
  }
};


export const resetPassword = async (data: { 
  usr_nombre: string; 
  usr_respuesta: string; 
  new_password: string; 
}) => {
  try {
    const cleanData = {
      ...data,
      usr_respuesta: data.usr_respuesta.toLowerCase().trim()
    };
    const res = await api.post("/security/reset-password", cleanData);
    return res.data;
  } catch (error: any) {
    return error.response?.data || { result: false, message: "Error al restablecer contraseña" };
  }
};

File: src/api/Services/recovery_service.py
from flask_restful import Resource
from flask import request
from ..Components.recovery_component import RecoveryComponent
from ...utils.general.response import response_error, response_success
from .middleware import valida_api_token


class ChangePasswordService(Resource):
    @valida_api_token
    def post(self):
        """ Cambiar clave estando logueado """
        data = request.get_json()
        user_id = data.get('usr_id')
        old_p = data.get('old_password')
        new_p = data.get('new_password')

        if not all([user_id, old_p, new_p]):
            return response_error("Faltan campos obligatorios")

        res = RecoveryComponent.changePassword(user_id, old_p, new_p)
        if res['result']:
            return response_success("Contraseña actualizada exitosamente")
        return response_error(res['message'])

class ResetPasswordService(Resource):
    def post(self):
        """ Recuperar clave olvidada """
        data = request.get_json()
        login = data.get('usr_nombre')
        respuesta = data.get('usr_respuesta')
        new_p = data.get('new_password')
        # Aquí idealmente recibirías un 'token_de_correo'

        if not login or not new_p or not respuesta:
            return response_error("Login, respuesta y nueva clave son requeridos")

        res = RecoveryComponent.resetPassword(login, respuesta, new_p)
        if res['result']:
            return response_success(None, "Se ha restablecido la contraseña")
        return response_error(res['message'])
